### PR TITLE
[feat] added optional parameter which can toggle on/off searching while typing

### PR DESCRIPTION
**Pull request by thaddmt**
- Added option `showResultsWhileTyping` (default: false)
  - When false requires enter key press to perform search
  - When true will automatically perform search when string is above `minLength`

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] run `npm run docs` and commit changes to API.md
 - [x] update CHANGELOG.md with changes under `master` heading before merging

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -11,118 +11,120 @@
...
-    -   [setFilter][46]
-        -   [Parameters][47]
-    -   [setGeocoderApi][48]
-        -   [Parameters][49]
-    -   [getGeocoderApi][50]
-    -   [on][51]
-        -   [Parameters][52]
-    -   [off][53]
-        -   [Parameters][54]
+    -   [\_fitBoundsForMarkers][10]
```
Comment (Amplifiyer) [inline API.md:14]: are these trailing characters `\_` intentional since it kind of signal that it's private but part of API.md?

File: API.md [hunk 2]
```diff
@@ -11,118 +11,120 @@
...
+    -   `options.getItemValue` **[Function][66]?** A function that specifies how the selected result should be rendered in the search bar. This function should accept a single [Carmen GeoJSON][67] object as input and return a string. HTML tags in the output string will not be rendered. Defaults to `(item) => item.place_name`.
+    -   `options.localGeocoderOnly` **[Boolean][60]** If `true`, indicates that the `localGeocoder` results should be the only ones returned to the user. If `false`, indicates that the `localGeocoder` results should be combined with those from the Maplibre API with the `localGeocoder` results ranked higher. (optional, default `false`)
+    -   `options.showResultsWhileTyping` **[Boolean][60]** If `fales`, indicates that search will only occur on enter key press. If `false`, indicates that the Geocoder will search on the input box being updated above the minLength option. (optional, default `false`)
```
Comment (Amplifiyer) [inline API.md:106]:
```suggestion
    -   `options.showResultsWhileTyping` **[Boolean][60]** If `false`, indicates that search will only occur on enter key press. If `true`, indicates that the Geocoder will search on the input box being updated above the minLength option. (optional, default `false`)
```

File: API.md [hunk 3]
```diff
@@ -140,67 +142,71 @@ geocoder.addTo('#geocoder-container');
...
 
--   `searchInput` **[string][62]** location name or other search input
+-   `searchInput` **[string][63]** location name or other search input
+
+Returns **[MaplibreGeocoder][69]** this
 
-Returns **[MaplibreGeocoder][68]** this
+### \_fitBoundsForMarkers
+
+Fits the map to the current bounds for the searched results
```
Comment (Amplifiyer) [inline API.md:167]: This API doesn't return `this`?

File: lib/index.js [hunk 4]
```diff
@@ -50,11 +50,12 @@ var subtag = require('subtag');
...
  * @param {Function} [options.getItemValue] A function that specifies how the selected result should be rendered in the search bar. This function should accept a single [Carmen GeoJSON](https://github.com/mapbox/carmen/blob/master/carmen-geojson.md) object as input and return a string. HTML tags in the output string will not be rendered. Defaults to `(item) => item.place_name`.
  * @param {Boolean} [options.localGeocoderOnly=false] If `true`, indicates that the `localGeocoder` results should be the only ones returned to the user. If `false`, indicates that the `localGeocoder` results should be combined with those from the Maplibre API with the `localGeocoder` results ranked higher.
+ * @param {Boolean} [options.showResultsWhileTyping=false] If `fales`, indicates that search will only occur on enter key press. If `false`, indicates that the Geocoder will search on the input box being updated above the minLength option.
```
Comment (Amplifiyer) [inline lib/index.js:53]:
```suggestion
 * @param {Boolean} [options.showResultsWhileTyping=false] If `false`, indicates that search will only occur on enter key press. If `true`, indicates that the Geocoder will search on the input box being updated above the minLength option.
```

File: lib/index.js [hunk 5]
```diff
@@ -373,37 +378,23 @@ MaplibreGeocoder.prototype = {
...
-        }
-
-        if (results.length > 0 && this._maplibregl) {
-          this._handleResultMarkers(results);
+      if (!this.options.showResultsWhileTyping) {
+        this._geocode(target.value);
+      } else {
+        if (this.options.showResultMarkers) {
+          this._fitBoundsForMarkers();
         }
```
Comment (Amplifiyer) [inline lib/index.js:386]: Lacking some context on this change here, why `this._fitBoundsForMarkers()` is not applicable here when `showResultsWhileTyping==false` and rather moved to lines 641,642.

File: API.md (hunk 1 above)
Comment (thaddmt) [reply API.md:14]: Whoops forgot to add a private tag so this wouldnt show up

File: lib/index.js (hunk 5 above)
Comment (thaddmt) [reply lib/index.js:386]: With the new feature when `showResultsWhileTyping` is false we will only show results after pressing enter. In that case we also only want to attempt to `_fitBoundsForMarkers()` when showing results which is done in the `_geocode()` method (lines 641,642).

When `showResultsWhileTyping` is true then `geocode()` method is being automatically called while typing, already showing results in the field. In that case when users press enter they are either pressing enter on the text field to indicate they want to `_fitBoundsForMarkers()` or pressing enter (or clicking on) a specific result indicating they want to `flyTo` a single marker

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**